src/Infrastructure/Controller/Regulation/AbstractRegulationController.php

```diff
@@ -37,24 +37,26 @@ protected function getRegulationOrderRecordUsing(callable $func): mixed
             throw new NotFoundHttpException();
         }
 
-        /** @var SymfonyUser */
-        $user = $this->security->getUser();
+        if ($mustValidateOrganization) {
+            /** @var SymfonyUser|null */
+            $user = $this->security->getUser();
 
-        if (!$this->canOrganizationAccessToRegulation->isSatisfiedBy($regulationOrderRecord, $user->getOrganizationUuids())) {
-            throw new AccessDeniedHttpException();
+            if ($user && !$this->canOrganizationAccessToRegulation->isSatisfiedBy($regulationOrderRecord, $user->getOrganizationUuids())) {
```

Comment:
Selon moi c'est plutôt `if (!$user || !$this->...())`, non ?

Si on exige `mustValidateOrg` mais qu'on n'est pas connecté il faut lever l'exception AccessDenied aussi ?

Reply:
En fait pour la lisibilité peut-être qu'il vaut mieux explicitement faire ce check en dehors de cette méthode abstraite. Le "true" par défaut défaut rend le code plus difficile à saisir je trouve (au contraire d'un "false" par défaut qui ferait qu'on déclenche ce check explicitement dans le controller).

Par ailleurs jusqu'ici on a fait tous les checks de permissions / specs explicitement dans les controllers donc ça me paraît bien de continuer comme ça.

Reply:
J'ai voulu garder ce fonctionnement centra car cette méthode est utilisée à différents endroits (suppression, duplication, création d'un arrêté, etc). 
Ça évite justement de devoir déporter cette logique unitairement dans tous ces controlleurs.
Il n'y a maintenant plus que dans le controlleur `RegulationDetailControlleur` qu'on a plus besoin de valider à ce niveau l'organisation.

Reply:
Peut-être faire l'inverse alors

Mettre

```php
requireUserSameOrg: true
```

Dans tous les controllers où on veut faire la validation ?

(Je propose aussi un renommage car "validate" ça n'indique pas vraiment dans quel sens on valide)

Reply:
D'accord avec le côté ambiguë du terme "validate"

Reply:
Ok pour le terme `requireUserSameOrg`, moins pour faire inverser la vérification. Je préfère avoir une approche restrictive par défaut pour ne pas exposer un point d'entrée si jamais on oublie de le faire à un endroit.